spring-integration-file/src/main/java/org/springframework/integration/file/dsl/RemoteFileOutboundGatewaySpec.java

```diff
@@ -0,0 +1,175 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.file.dsl;
+
+import java.io.File;
+import java.util.function.Function;
+
+import org.springframework.expression.Expression;
+import org.springframework.integration.dsl.MessageHandlerSpec;
+import org.springframework.integration.expression.FunctionExpression;
+import org.springframework.integration.file.filters.CompositeFileListFilter;
+import org.springframework.integration.file.filters.FileListFilter;
+import org.springframework.integration.file.filters.RegexPatternFileListFilter;
+import org.springframework.integration.file.filters.SimplePatternFileListFilter;
+import org.springframework.integration.file.remote.gateway.AbstractRemoteFileOutboundGateway;
+import org.springframework.messaging.Message;
+import org.springframework.util.Assert;
+
+/**
+ *
+ * @param <F> the target file type.
+ * @param <S> the target {@link RemoteFileOutboundGatewaySpec} implementation type.
+ *
+ * @author Artem Bilan
+ *
+ * @since 5.0
+ */
+public abstract class RemoteFileOutboundGatewaySpec<F, S extends RemoteFileOutboundGatewaySpec<F, S>>
+		extends MessageHandlerSpec<S, AbstractRemoteFileOutboundGateway<F>> {
+
+	private CompositeFileListFilter<F> filter;
+
+	private CompositeFileListFilter<File> mputFilter;
+
+	protected RemoteFileOutboundGatewaySpec(AbstractRemoteFileOutboundGateway<F> outboundGateway) {
+		this.target = outboundGateway;
+		this.target.setRequiresReply(true);
+	}
+
+	public S options(String options) {
+		this.target.setOptions(options);
+		return _this();
+	}
+
+	public S options(AbstractRemoteFileOutboundGateway.Option... options) {
+		Assert.noNullElements(options);
+		StringBuilder optionsString = new StringBuilder();
+		for (AbstractRemoteFileOutboundGateway.Option option : options) {
+			optionsString.append(option.getOption()).append(" ");
+		}
+		this.target.setOptions(optionsString.toString());
+		return _this();
+	}
+
+	public S remoteFileSeparator(String remoteFileSeparator) {
+		this.target.setRemoteFileSeparator(remoteFileSeparator);
+		return _this();
+	}
+
+	public S localDirectory(File localDirectory) {
+		this.target.setLocalDirectory(localDirectory);
+		return _this();
+	}
+
+	public S localDirectoryExpression(String localDirectoryExpression) {
+		return localDirectoryExpression(PARSER.parseExpression(localDirectoryExpression));
+	}
+
+	public <P> S localDirectory(Function<Message<P>, String> localDirectoryFunction) {
+		return localDirectoryExpression(new FunctionExpression<>(localDirectoryFunction));
+	}
+
+	/**
+	 * @param localDirectoryExpression a SpEL expression to evaluate the local directory.
+	 * @return the Spec.
+	 */
+	public S localDirectoryExpression(Expression localDirectoryExpression) {
```

Comment:
Np. I'm on it.
Addressing all the missed JavaDocs in these DSL classes for File module...